### PR TITLE
Adopt an open governance model

### PR DESCRIPTION
**Pull request by stefanalund**
## Hey, I just made a Pull Request!

As part of [donating](https://github.com/cncf/sandbox/issues/274) Backstage the CNCF we are adopting an open governance model:

https://github.com/opengovernance/opengovernance.dev

This PR adds a first revision of the Backstage governance model. It borrows from other CNCF projects like:

https://github.com/cortexproject/cortex/blob/master/GOVERNANCE.md
https://github.com/linkerd/linkerd2/blob/main/GOVERNANCE.md

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/spotify/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: MAINTAINERS.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+## Plugin maintainers 🧩
+
+Teams and individuals that maintain a plugin (or another non-core module of the code) can get write access to that part of the repo using CODEOWNERS.
+
+## Hall of Fame 👏
+
+People that have made significant contributions to the project and earned write access:
+
+- Andrew Thauer - Wealthsimple (GitHub: @andrewthauer)
+- Oliver Sand - SDA-SE (GitHub: @Fox32)
```
Comment (Fox32) [inline MAINTAINERS.md:21]:
```suggestion
- Oliver Sand - SDA SE (GitHub: @Fox32)
```
Just a minor change, but that's the right company name 😉

**Review by Rugvip — APPROVED** [review]
🎉

File: GOVERNANCE.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+- understanding of how the team works (policies, processes for testing and code review, etc),
+- understanding of the project's code base and coding style.
+
+## Changes in Maintainership
+
+A new maintainer must be proposed by an existing maintainer by opening an issue (with title `Maintainer Nomination`) to the Backstage GitUub repository (https://github.com/backstage/backstage) containing the following information:
+
+- nominee's first and last name,
+- nominee's email address and GitHub user name,
+- an explanation of why the nominee should be a committer,
```
Comment (adamdmharvey) [inline GOVERNANCE.md:32]:
```suggestion
- an explanation of why the nominee should be a maintainer,
```

maybe?

**Review by adamdmharvey — COMMENTED** [review]
🎉

File: GOVERNANCE.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  - perform code reviews for 10 non-trivial pull requests,
+  - contribute 10 non-trivial pull requests and have them merged into master,
+- ability to write good code,
+- ability to collaborate with the team,
+- understanding of how the team works (policies, processes for testing and code review, etc),
+- understanding of the project's code base and coding style.
+
+## Changes in Maintainership
+
+A new maintainer must be proposed by an existing maintainer by opening an issue (with title `Maintainer Nomination`) to the Backstage GitUub repository (https://github.com/backstage/backstage) containing the following information:
```
Comment (andrewthauer) [inline GOVERNANCE.md:28]:
```suggestion
A new maintainer must be proposed by an existing maintainer by opening an issue (with title `Maintainer Nomination`) to the Backstage GitHub repository (https://github.com/backstage/backstage) containing the following information:
```

**Review by andrewthauer — DISMISSED** [review]
:tada: Small typo

**Pull request merged**